src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -734,14 +784,108 @@ subroutine read_fv3_netcdf_guess(fv3filenamegin)
       call gsi_fv3ncdf_read_v1(tracers,'o3mr','O3MR',ges_oz,mype_oz)
     endif
 
+    if (l_use_dbz_directDA ) then
+      if( fv3sar_bg_opt == 0) then
+        call gsi_fv3ncdf_read(dynvars,'W','w',ges_w,mype_w)
+        call gsi_fv3ncdf_read(tracers,'LIQ_WAT','liq_wat',ges_ql,mype_ql)
+        call gsi_fv3ncdf_read(tracers,'ICE_WAT','ice_wat',ges_qi,mype_qi)
+        call gsi_fv3ncdf_read(tracers,'RAINWAT','rainwat',ges_qr,mype_qr)
+        call gsi_fv3ncdf_read(tracers,'SNOWWAT','snowwat',ges_qs,mype_qs)
+        call gsi_fv3ncdf_read(tracers,'GRAUPEL','graupel',ges_qg,mype_qg)
+        call gsi_fv3ncdf_read(tracers,'RAIN_NC','rain_nc',ges_qnr,mype_qnr)
+      else
+         write(6,*) "FV3 IO READ for 'fv3sar_bg_opt == 0 are only available for now in direct reflectivity DA"
+         stop
+      end if
+
 !!  tsen2tv  !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-    do k=1,nsig
-       do j=1,lon2
-          do i=1,lat2
-             ges_tv(i,j,k)=ges_tsen(i,j,k,it)*(one+fv*ges_q(i,j,k))
+      do k=1,nsig
+        do i=1,lon2
+          do j=1,lat2
+             ges_tv(j,i,k)=ges_tsen(j,i,k,it)*(one+fv*ges_q(j,i,k))
+!            Apply power transform if option is ON 
+             if (l_use_cvpqx) then
+                if ( l_use_cvpqx_pval .eq. 0. ) then ! CVlogq
+                   if (ges_tsen(j,i,k,it) .gt. 274.15) then
```

Comment:
This part of the code to setup qr_min, qr_thrshd, qs_min, qs_thrshd, qg_min, qg_thershd. It appears in several locations. Can you convert this part to a function or subroutine to setup those variables? Adding function could make code run slower. If GSI use inline compiling, adding function should be OK.

Reply:
I created to subroutines according to your comment since these codes also exist in WRF code. I replaced these parts with the subroutines in ‘gsi_rfv3io_mod.f90’ and ‘cplr_read_wrf_mass_guess.f90’ and ‘cplr_wrwrfmassa.f90’.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-4c910ebd862ea8bea4024bbfa73835e643efb4091155d4e8ef121388b14ab6ffR2744-R3211